Ignore unknown provider specific properties

kevinfrommelt opened the pull request:
**Description**

Currently if an endpoint contains a provider specific property for a provider other than the currently configured provider, the current provider will still consider it in the desired state. This will cause the endpoint to be marked as needing an update because when the provider loads the actual state it will never have that specific property set (ie. the desired and actual state never match). This causes unnecessary updates to be made on every sync.

This PR updates the providers to only consider its own specific properties (ie. ignore properties for other providers).

**Checklist**

- [X] Unit tests updated
- [ ] End user documentation updated

mloiseleur commented:
/ok-to-test

mloiseleur commented:
Mmmh :thinking: I guess it's also the case for other providers.
@kevinfrommelt  Do you think you can update this PR to solve this for all providers ?

kevinfrommelt commented:
@mloiseleur Yea I can give it a shot.

kevinfrommelt commented:
@mloiseleur Before I go through every provider, what do you think of this approach? 
https://github.com/kubernetes-sigs/external-dns/compare/master...kevinfrommelt:external-dns:ignore-other-provider-properties
It applies a filter to the BaseProvider, which can be overridden by other providers.

mloiseleur commented:
At first glance, it should do it :+1: !

kevinfrommelt commented:
@mloiseleur I've updated this to cover all providers. This ready for another review.

mloiseleur commented:
@kevinfrommelt You'll need to add test for `endpoint/provider_specific_property_filter.go`

Btw, wdyt of logging a warning when there is property not related to the provider ? Is there legitimate / expected use cases where it happens ?

kevinfrommelt commented:
Hey @mloiseleur, sorry for delay. I have rebased onto latest, added tests for `endpoint/provider_specific_property_filter.go`, and added a debug log for properties that get ignored by the provider.

> is there legitimate / expected use cases where it happens ?

As far as I could tell the providers were already ignoring all the properties they didn't have defined. The extra properties were just throwing off the comparison between the desired and actual states of the endpoints, which was triggering the provider to do a no-op update on every sync.

kevinfrommelt commented:
@mloiseleur Just checking in to see if you've had a chance to look at this again.

ivankatliarchuk commented on endpoint/provider_specific_property_filter.go:
```diff
@@ -0,0 +1,61 @@
+/*
```
Could we move this instead to `pkg/filters` ? Example pull request https://github.com/gofogo/k8s-sigs-external-dns-fork/pull/3/files, is not officially a pull request against this repository, as there is another work that needs doing as well

ivankatliarchuk reviewed (changes requested):
Have you tested all this providers, would you be able to share a set of manual test steps using manifests and kubectl commands? This will help in understanding and verifying the implementation.

ivankatliarchuk commented:
Hi @kevinfrommelt  are you still interested in this pull request? If yes, pls address the review.

We also need a set of manual test steps using manifests and kubectl commands. Example with verification https://github.com/kubernetes-sigs/external-dns/pull/5085#issuecomment-2651003555

mloiseleur commented:
There is no answer from PR author, so I'll close this PR.
Feel free to re-open or open a new one if you need it.
